core/variable_operations.cpp

```diff
@@ -264,36 +264,110 @@ Variable reverse(Variable var, const Dim dim) {
+                              const VariableView &out) {
+
   transform_in_place<std::tuple<double, float>>(
       out, var, replacement,
       scipp::overloaded{
           transform_flags::expect_all_or_none_have_variance,
-          [](auto &a, const auto &b, const auto &repl) {
-            a = isnan(b) ? repl : b;
-          },
+          [&op](auto &a, const auto &b, const auto &repl) { op(a, b, repl); },
           [](units::Unit &a, const units::Unit &b, const units::Unit &repl) {
             expect::equals(b, repl);
             a = b;
           }});
   return out;
 }
 
-Variable nan_to_num(const VariableConstView &var,
-                    const VariableConstView &replacement) {
-  using std::isnan;
+template <typename Op>
+Variable replace(const VariableConstView &var,
+                 const VariableConstView &replacement, Op op) {
   return transform<std::tuple<double, float>>(
       var, replacement,
-      overloaded{
-          transform_flags::expect_all_or_none_have_variance,
-          [](const auto &x, const auto &repl) { return isnan(x) ? repl : x; },
-          [](const units::Unit &x, const units::Unit &repl) {
-            expect::equals(x, repl);
-            return x;
-          }});
+      overloaded{transform_flags::expect_all_or_none_have_variance,
+                 [&op](const auto &x, const auto &repl) { return op(x, repl); },
+                 [](const units::Unit &x, const units::Unit &repl) {
+                   expect::equals(x, repl);
+                   return x;
+                 }});
+}
+} // namespace
+
+VariableView nan_to_num(const VariableConstView &var,
+                        const VariableConstView &replacement,
+                        const VariableView &out) {
+
+  auto op = [](auto &a, const auto &b, const auto &repl) {
+    if constexpr (is_ValueAndVariance_v<std::decay_t<decltype(b)>>)
+      a = isnan(b) ? repl : b;
+    else
+      a = std::isnan(b) ? repl : b;
+  };
+
+  return replace_in_place(var, replacement, op, out);
+}
+
+VariableView positive_inf_to_num(const VariableConstView &var,
+                                 const VariableConstView &replacement,
+                                 const VariableView &out) {
+  auto op = [](auto &a, const auto &b, const auto &repl) {
+    if constexpr (is_ValueAndVariance_v<std::decay_t<decltype(b)>>)
+      a = isinf(b) && b.value > 0 ? repl : b;
+    else
+      a = std::isinf(b) && b > 0 ? repl : b;
+  };
+
+  return replace_in_place(var, replacement, op, out);
+}
+VariableView negative_inf_to_num(const VariableConstView &var,
+                                 const VariableConstView &replacement,
+                                 const VariableView &out) {
+  auto op = [](auto &a, const auto &b, const auto &repl) {
+    if constexpr (is_ValueAndVariance_v<std::decay_t<decltype(b)>>)
+      a = isinf(b) && b.value < 0 ? repl : b;
+    else
+      a = std::isinf(b) && b < 0 ? repl : b;
```

Comment:
See above.

Reply:
This would be fine for if we only had to check for `inf` (ignoring sign) but overloading `>` and `<` on `ValueAndVariance` would need to be added. This should also account for uncertainties when performing the inequality.

Reply:
Right, missed the `.value` in there 👍